cloud/services/compute/instances.go

```diff
@@ -146,6 +150,32 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*compute.Instance,
 	if scope.GCPMachine.Spec.RootDeviceType != nil {
 		input.Disks[0].InitializeParams.DiskType = diskTypeURL(scope.Zone(), *scope.GCPMachine.Spec.RootDeviceType)
 	}
+	for _, d := range scope.GCPMachine.Spec.AdditionalDisks {
+		ad := &compute.AttachedDisk{
+			AutoDelete: true,
+			InitializeParams: &compute.AttachedDiskInitializeParams{
+				DiskSizeGb: defaultDiskSizeGB,
+				DiskType:   diskTypeURL(scope.Zone(), infrav1.PdStandardDiskType),
+			},
+		}
+		if d.DeviceType != nil {
+			ad.InitializeParams.DiskType = diskTypeURL(scope.Zone(), *d.DeviceType)
+		}
+		if *d.DeviceType == infrav1.LocalSsdDiskType {
+			ad.Type = "SCRATCH" // Default is PERSISTENT.
+			ad.InitializeParams.DiskSizeGb = 375
+
+			// For local SSDs set interface to NVME (instead of default SCSI) which is faster.
+			// Most OS images would work with both NVME and SCSI disks but some may work
+			// considerably faster with NVME.
+			// https://cloud.google.com/compute/docs/disks/local-ssd#choose_an_interface
+			ad.Interface = "NVME"
+		} else if d.Size > 0 {
+			ad.InitializeParams.DiskSizeGb = d.Size
+		}
+
+		input.Disks = append(input.Disks, ad)
```

Comment:
```suggestion
		ad := &compute.AttachedDisk{
			AutoDelete: true,
			InitializeParams: &compute.AttachedDiskInitializeParams{
				DiskSizeGb: d.Size,
				DiskType:   diskTypeURL(scope.Zone(), pointer.StringPtrDerefOr(d.DeviceType, infrav1.PdStandardDiskType)),
			},
		}
		
		if ad.InitializeParams.DiskSizeGb == 0 {
			ad.InitializeParams.DiskSizeGb = defaultDiskSizeGB
		}
		
		if ad.InitializeParams.DiskType == infrav1.LocalSsdDiskType{
			ad.Type = "SCRATCH" // Default is PERSISTENT.
			
			// Override the Disk size
			ad.InitializeParams.DiskSizeGb = 375

			// For local SSDs set interface to NVME (instead of default SCSI) which is faster.
			// Most OS images would work with both NVME and SCSI disks but some may work
			// considerably faster with NVME.
			// https://cloud.google.com/compute/docs/disks/local-ssd#choose_an_interface
			ad.Interface = "NVME"
		}

		input.Disks = append(input.Disks, ad)
```